app/src/main/java/com/example/testjavaapp/fragments/NotificationFragment.java

```diff
@@ -0,0 +1,131 @@
+package com.example.testjavaapp.fragments;
+
+
+import android.app.NotificationChannel;
+import android.app.NotificationManager;
+import android.app.PendingIntent;
+import android.content.Intent;
+import android.os.Build;
+import android.os.Bundle;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.annotation.RequiresApi;
+import androidx.core.app.NotificationCompat;
+import androidx.core.app.NotificationManagerCompat;
+import androidx.fragment.app.Fragment;
+import androidx.navigation.NavDeepLinkBuilder;
+
+import android.service.notification.StatusBarNotification;
+import android.util.Log;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.Button;
+
+import com.example.testjavaapp.MainActivity;
+import com.example.testjavaapp.R;
+
+import static android.content.Context.NOTIFICATION_SERVICE;
+
+
+/**
+ * A simple {@link Fragment} subclass.
+ */
+public class NotificationFragment extends Fragment {
+
```

Comment:
Also, I can't see any data exchange between Activities. Please do

Reply:
Done. I pass String from MainActivity to ServiceActivity. Similar approach was implemented when clicked on notification that passed String to MainActivity